packages/contracts/src/core/permission/PermissionManager.sol

```diff
@@ -164,9 +444,22 @@ abstract contract PermissionManager is Initializable {
     function applySingleTargetPermissions(
         address _where,
         PermissionLib.SingleTargetPermission[] calldata items
-    ) external virtual auth(ROOT_PERMISSION_ID) {
+    ) external virtual auth(APPLY_TARGET_PERMISSION_ID) { // TODO: Check types here 
         for (uint256 i; i < items.length; ) {
             PermissionLib.SingleTargetPermission memory item = items[i];
+            Permission storage permission = permissions[permissionHash(_where, item.permissionId)];
+
+            if (
+                permission.created &&
+                !_checkPermissionsForApplyTargetMethods(
+                    permission,
+                    _where,
+                    item.permissionId,
+                    item.operation
+                )
+            ) {
+                revert InvalidDelegation(msg.sender, _where, item.permissionId);
```

Comment:
The above function also checks if the caller is a `owner`. So maybe a better error name: "OwnerOrDelegateNeither` - name is horrible, but you get the idea.